inst/htmljs/animint.js

```diff
@@ -512,76 +512,82 @@ var animint = function (to_select, json_file) {
-        });
+      if(Object.keys(p_info.panel_background).length > 1) {
+        svg.insert("rect", "#plottitle")
+          .attr("x", plotdim.xstart)
+          .attr("y", plotdim.ystart)
+          .attr("width", plotdim.xend - plotdim.xstart)
+          .attr("height", plotdim.yend - plotdim.ystart)
+          .attr("class", "background_rect")
+          .style("fill", p_info.panel_background.fill)
+          .style("stroke", p_info.panel_background.colour)
+          .style("stroke-dasharray", function() {
+            return linetypesize2dasharray(p_info.panel_background.linetype,
+                                          p_info.panel_background.size);
+          });
+      }
       
       // function to draw the grid lines when supplied with one of the elements
       var grid_line = function(grid_background, grid_class) {
-        var col = grid_background.colour;
-        var lt = grid_background.linetype;
-        var size = grid_background.size;
-        var cap = grid_background.lineend;
-
-        // draw horizontal grid lines if they are defined
-        if(typeof grid_background.loc.y != "undefined") {
-          var draw_hor_line = function(element) {
-            svg.insert("line", "#plottitle")
-              .attr("x1", plotdim.xstart)
-              .attr("x2", plotdim.xend)
-              .attr("y1", function() { return scales[panel_i].y(element); })
-              .attr("y2", function() { return scales[panel_i].y(element); })
-              .attr("class", function() { return "grid " + grid_class; })
-              .style("stroke", col)
-              .style("stroke-linecap", cap)
-              .style("stroke-width", size)
-              .style("stroke-dasharray", function() {
-                return linetypesize2dasharray(lt, size);
-              });
+        if(Object.keys(grid_background).length > 1) {
+          var col = grid_background.colour;
+          var lt = grid_background.linetype;
+          var size = grid_background.size;
+          var cap = grid_background.lineend;
+
+          // draw horizontal grid lines if they are defined
+          if(typeof grid_background.loc.y != "undefined") {
+            var draw_hor_line = function(element) {
+              svg.insert("line", "#plottitle")
+                .attr("x1", plotdim.xstart)
+                .attr("x2", plotdim.xend)
+                .attr("y1", function() { return scales[panel_i].y(element); })
+                .attr("y2", function() { return scales[panel_i].y(element); })
+                .attr("class", function() { return "grid " + grid_class; })
+                .style("stroke", col)
+                .style("stroke-linecap", cap)
+                .style("stroke-width", size)
+                .style("stroke-dasharray", function() {
+                  return linetypesize2dasharray(lt, size);
+                });
+            }
+            grid_background.loc.y.forEach(draw_hor_line);
```

Comment:
To add grid lines you used forEach, which is fine, but I would have used a d3.js data-bind. The advantage of using a data-bind is that the elements would be easier to update. However that doesn't really matter for grid lines since they are added once at the beginning of plot rendering and then they do not change. Anyways if you are interested in learning about d3 data-binds I would suggest reading https://github.com/mbostock/d3/wiki/Selections#d3_selectAll and http://bost.ocks.org/mike/circles/